src/storage.rs

```diff
@@ -92,19 +93,39 @@ pub struct Eip7702Code {
     version: u8,
 }
 
+/// EVM Object Format (EOF) code.
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct EofCode {
+    /// EOF header.
+    pub header: EofHeader,
+    /// EOF body.
+    pub body: EofBody,
+}
+
 /// EVM Code, currently mapping to REVM's [ByteCode].
-// TODO: Support EOF
 #[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
 pub enum EvmCode {
     /// Maps both analyzed and non-analyzed REVM legacy bytecode.
     Legacy(LegacyCode),
     /// Maps delegated EIP7702 bytecode.
     Eip7702(Eip7702Code),
+    /// Maps EOF bytecode.
+    Eof(EofCode),
 }
 
 impl From<EvmCode> for Bytecode {
     fn from(code: EvmCode) -> Self {
         match code {
+            EvmCode::Eof(eof_code) => {
```

Comment:
I think we should construct a `revm::EofBody` from our `EofCode` then call `revm::Eof::new` or `revm::EofBody::into_eof`.

Neat but can you also help re-order the arms here to `Legacy` -> `Eip7702` -> `Eof`?